functional-test/src/test/groovy/org/rundeck/tests/functional/api/export_import/JobsExportSpec.groovy

```diff
@@ -0,0 +1,237 @@
+                }
+            }
+            def countJobId = jobIdAux.tokenize().findAll { it == jobId } size()
+            return (countJobId == 1 && jobUuid == jobId) ?
+                    "OK" :
+                    "Wrong job count: ${countJobId}, or wrong found id: ${jobUuid}"
+        }
+    }
+
+    def validateYaml(String yamlContent) {
+        if (!yamlContent.contains(':') || !yamlContent.contains('\n')) {
+            return "ERROR: Response was not valid YAML"
+        }
+        def yamlMap = [:]
+        yamlContent.split('\n').each { line ->
+            def parts = line.trim().split(':', 2)
+            if (parts.size() == 2) {
+                yamlMap[parts[0].trim()] = parts[1].trim()
+            }
+        }
+        if (!yamlMap.containsKey('name')) {
+            return "ERROR: Response did not contain expected result"
+        }
+        String jobIdAux = yamlMap['id']
+        String jobUuid = yamlMap['uuid']
+        return (jobIdAux == jobId && jobUuid == jobId) ?
+                "OK" :
+                "Wrong job id: ${jobUuid}"
+    }
+
+    def "export RunDeck Jobs in jobs.json format"() {
+        when:
+            def export = doGet("/project/${PROJECT_NAME}/jobs/export?format=json")
+        then:
+            verifyAll {
+                def header = export.headers().get("Content-Type")
+                header != null
+                header == "application/json" || "application/json;charset=utf-8"
+                def json = jsonValue(export.body(), List)
+                json.size() > 0
+            }
+    }
+
+    def "export RunDeck Jobs in jobs default (json) format"() {
+        when:
+            def export = doGet("/project/${PROJECT_NAME}/jobs/export")
+        then:
+            verifyAll {
+                def header = export.headers().get("Content-Type")
+                header != null
+                header == "application/json" || "application/json;charset=utf-8"
+                def json = jsonValue(export.body(), List)
+                json.size() > 0
+            }
+    }
+
+    def "export RunDeck Jobs in jobs.xml format"() {
+        when:
+            def export = doGet("/project/${PROJECT_NAME}/jobs/export?format=xml")
+        then:
+        verifyAll {
+            def header = export.headers().get("Content-Type")
+            header != null
+            header == "text/xml" || "text/xml;charset=utf-8"
+            def xml = export.body().string()
+            xml.contains("<joblist>")
+            xml.count("<job>") > 0
+        }
+    }
+
+    def "export RunDeck Jobs in jobs.yaml format"() {
+        when:
+            def export = doGet("/project/${PROJECT_NAME}/jobs/export?format=yaml")
+        then:
+            verifyAll {
+                def header = export.headers().get("Content-Type")
+                header != null
+                header == "text/yaml" || "text/yaml;charset=utf-8"
+                def xml = export.body().string()
```

Comment:
shouldn't this be an yml and be validated as such?

Reply:
The original test only validates that the response header is “Content-Type: text/yaml”